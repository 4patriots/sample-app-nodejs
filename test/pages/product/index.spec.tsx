import { useRouter } from "next/router";
import SidePanel from "@pages/product/[productId]"; 
import { render, screen} from '@test/utils';

jest.mock('@lib/hooks', () => require('@mocks/hooks'));
jest.mock('next/router', () => ({
    useRouter: jest.fn(),
}));

describe('SidePanel', () => {
    const router = { query: { pid: '1' } };
    useRouter.mockReturnValue(router);

    test('renders correctly', async () => {
        const { container } = render(<SidePanel />);
        // Wait for table to be rendered
        screen.getByText('Basic Information');

        expect(container.firstChild).toMatchSnapshot();
    });
});

