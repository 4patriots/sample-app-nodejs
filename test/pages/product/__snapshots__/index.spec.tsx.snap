// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidePanel renders correctly 1`] = `
<div
  class="styled__StyledBox-sc-sj5f1m-0 odkmb styled__StyledPanel-sc-1h6ef3q-0 gyZFEm"
>
  <div
    class="styled__StyledBox-sc-sj5f1m-0 gHBxjL styled__StyledFlex-sc-hcvk8l-0 hPcIGA"
  >
    <h2
      class="styled__StyledH2-sc-tqnj75-2 styled__StyledH2-sc-1h6ef3q-1 kEGvNd fjdBDH"
    >
      Basic Information
    </h2>
  </div>
  <p
    class="styled__StyledText-sc-tqnj75-5 dBDsjF"
  >
    Product name
  </p>
  <p>
    Product 1
  </p>
  <p
    class="styled__StyledText-sc-tqnj75-5 dBDsjF"
  >
    Product type
  </p>
  <p>
    Physical
  </p>
  <p
    class="styled__StyledText-sc-tqnj75-5 dBDsjF"
  >
    Default price (excluding tax)
  </p>
  <p>
    $ 
    20
  </p>
  <p />
</div>
`;
